pkg/apis/flowcontrol/types.go

```diff
@@ -265,6 +266,15 @@ type ResourcePolicyRule struct {
 	// +optional
 	// +listType=set
 	Namespaces []string
+
+	// `resourceNames` is a list of target resource names that restricts
+	// matches. A (get, update, patch, delete) request that specifies
+	// a target name matches if either (a) this list contains that target
+	// names or (b) this list contains "*".
+	// If not set, this list will be defaulted to ["*"].
+	// +optional
+	// +listType=set
+	ResourceNames []string
```

Comment:
I think this is actually the names of *objects* not the names of resources, which already appears above? I think maybe we should merge a beta version of the existing types without this additional concept, it's OK to add this field to both alpha and beta in a future change that doesn't have to happen in a time-sensitive window.